File: LocaCraft-app/src/app/models/lessor.ts
import { Lease } from "./lease";

export interface Lessor{
    id: number;
    firstName: string;
    lastName: string;
    address: string;
    city: string;
    postalCode: string;
    country: string;
    phone: string;
    email: string;
    leases: Lease[];
}

File: LocaCraftAPI/LocaCraftAPI/Models/Lessor.cs
using System.ComponentModel.DataAnnotations;

namespace LocaCraftAPI.Models
{
    public class Lessor
    {
        public int Id { get; set; }

        [Required]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        public string LastName { get; set; } = string.Empty;
        [Required]
        public string Address { get; set; } = string.Empty;
        [Required]
        public string City { get; set; } = string.Empty;
        [Required]
        public string PostalCode { get; set; } = string.Empty;
        [Required]
        public string Country { get; set; } = string.Empty;

        public string Phone { get; set; } = string.Empty;

        public string Email { get; set; } = string.Empty;

        public List<Lease> Leases { get; set; } = new();
    }
}
